moto/s3/models.py

```diff
@@ -1876,13 +1884,25 @@ def create_bucket(self, bucket_name: str, region_name: str) -> FakeBucket:
 
         return new_bucket
 
+    def create_table_storage_bucket(self, region_name: str) -> FakeTableStorageBucket:
+        # every s3 table is assigned a unique s3 bucket with a random name
+        bucket_name = f"{str(uuid4())}--table-s3"
```

Comment:
We should use `random.uuid4` here, to ensure the names are deterministic ([see the docs](https://docs.getmoto.org/en/latest/docs/configuration/recorder/index.html#deterministic-identifiers))